File: read_documents.py
"""
Script to read Word (.docx) and Excel (.xlsx) files
"""
import sys
from pathlib import Path

def read_docx(file_path):
    """Read Word document and return text content"""
    try:
        from docx import Document
        doc = Document(file_path)
        content = []
        
        for para in doc.paragraphs:
            if para.text.strip():
                content.append(para.text)
        
        # Also read tables if any
        for table in doc.tables:
            for row in table.rows:
                row_data = [cell.text.strip() for cell in row.cells]
                content.append(" | ".join(row_data))
        
        return "\n".join(content)
    except ImportError:
        return "Error: python-docx library not installed. Run: pip install python-docx"
    except Exception as e:
        return f"Error reading Word document: {str(e)}"

def read_xlsx(file_path):
    """Read Excel file and return content"""
    try:
        import openpyxl
        wb = openpyxl.load_workbook(file_path, data_only=True)
        content = []
        
        for sheet_name in wb.sheetnames:
            sheet = wb[sheet_name]
            content.append(f"\n{'='*50}")
            content.append(f"SHEET: {sheet_name}")
            content.append('='*50 + "\n")
            
            for row in sheet.iter_rows(values_only=True):
                # Skip completely empty rows
                if any(cell is not None and str(cell).strip() for cell in row):
                    row_str = " | ".join([str(cell) if cell is not None else "" for cell in row])
                    content.append(row_str)
        
        return "\n".join(content)
    except ImportError:
        return "Error: openpyxl library not installed. Run: pip install openpyxl"
    except Exception as e:
        return f"Error reading Excel document: {str(e)}"

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python read_documents.py <file_path>")
        sys.exit(1)
    
    file_path = sys.argv[1]
    file_ext = Path(file_path).suffix.lower()
    
    if file_ext == ".docx":
        content = read_docx(file_path)
    elif file_ext in [".xlsx", ".xls"]:
        content = read_xlsx(file_path)
    else:
        content = f"Unsupported file type: {file_ext}"
    
    print(content)

File: read_documents.js
const fs = require('fs');
const path = require('path');

async function readDocx(filePath) {
    try {
        const mammoth = require('mammoth');
        const result = await mammoth.extractRawText({ path: filePath });
        return result.value;
    } catch (error) {
        return `Error reading Word document: ${error.message}`;
    }
}

async function readXlsx(filePath) {
    try {
        const xlsx = require('xlsx');
        const workbook = xlsx.readFile(filePath);
        let content = [];
        
        workbook.SheetNames.forEach(sheetName => {
            content.push(`\n${'='.repeat(50)}`);
            content.push(`SHEET: ${sheetName}`);
            content.push('='.repeat(50) + '\n');
            
            const worksheet = workbook.Sheets[sheetName];
            const data = xlsx.utils.sheet_to_json(worksheet, { header: 1, defval: '' });
            
            data.forEach(row => {
                if (row.some(cell => cell !== '')) {
                    content.push(row.join(' | '));
                }
            });
        });
        
        return content.join('\n');
    } catch (error) {
        return `Error reading Excel document: ${error.message}`;
    }
}

async function main() {
    const filePath = process.argv[2];
    
    if (!filePath) {
        console.log('Usage: node read_documents.js <file_path>');
        process.exit(1);
    }
    
    const ext = path.extname(filePath).toLowerCase();
    
    let content;
    if (ext === '.docx') {
        content = await readDocx(filePath);
    } else if (ext === '.xlsx' || ext === '.xls') {
        content = await readXlsx(filePath);
    } else {
        content = `Unsupported file type: ${ext}`;
    }
    
    console.log(content);
}

main().catch(console.error);

